tests/commands/test_commands.py

```diff
@@ -411,3 +403,111 @@ def mock_write(tx, mode):
     resp = run_upsert_validator_new(args, b)
 
     assert b.get_transaction(resp)
+
+
+@pytest.mark.abci
+def test_upsert_validator_approve_with_tendermint(b, priv_validator_path, user_sk, validators):
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_new, \
+        run_upsert_validator_approve
+
+    public_key = '8eJ8q9ZQpReWyQT5aFCiwtZ5wDZC4eDnCen88p3tQ6ie'
+    new_args = Namespace(action='new',
+                         public_key=public_key,
+                         power=1,
+                         node_id='12345',
+                         sk=priv_validator_path,
+                         config={})
+
+    election_id = run_upsert_validator_new(new_args, b)
+
+    args = Namespace(action='approve',
+                     election_id=election_id,
+                     sk=priv_validator_path,
+                     config={})
+    approve = run_upsert_validator_approve(args, b)
+
+    assert b.get_transaction(approve)
+
+
+@pytest.mark.bdb
+@pytest.mark.tendermint
+def test_upsert_validator_approve_without_tendermint(b, priv_validator_path, new_validator, node_key):
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_approve
+    from argparse import Namespace
+
+    def mock_write(tx, mode):
+        b.store_bulk_transactions([tx])
+        return (202, '')
+
+    # patch the validator set. We now have one validator with power 10
+    b.get_validators = mock_get
+    b.write_transaction = mock_write
+
+    # our voters is a list of length 1, populated from our mocked validator
+    voters = ValidatorElection.recipients(b)
+    # and our voter is the public key from the voter list
+    voter = node_key.public_key
+    valid_election = ValidatorElection.generate([voter],
+                                                voters,
+                                                new_validator, None).sign([node_key.private_key])
+
+    # patch in an election with a vote issued to the user
+    election_id = valid_election.id
+    b.store_bulk_transactions([valid_election])
+
+    # call run_upsert_validator_approve with args that point to the election
+    args = Namespace(action='approve',
+                     election_id=election_id,
+                     sk=priv_validator_path,
+                     config={})
+
+    approval_id = run_upsert_validator_approve(args, b)
+
+    # assert returned id is in the db
+    assert b.get_transaction(approval_id)
+
+
+@pytest.mark.bdb
+@pytest.mark.tendermint
+def test_upsert_validator_approve_called_with_bad_key(b, bad_validator_path, new_validator, node_key):
+    from bigchaindb.commands.bigchaindb import run_upsert_validator_approve
+    from argparse import Namespace
+
+    def mock_write(tx, mode):
+        b.store_bulk_transactions([tx])
+        return (202, '')
+
+    # patch the validator set. We now have one validator with power 10
+    b.get_validators = mock_get
+    b.write_transaction = mock_write
```

Comment:
The code is being repeated in `test_upsert_validator_approve_without_tendermint` also. It would make sense to create a fixture and patch `b` and use it in both the functions.

Reply:
Done